src/components/ShareStructure/ShareStructure.vue

```diff
@@ -461,9 +462,8 @@ export default class ShareStructure extends Mixins(CommonMixin) {
     this.currentShareStructure.priority = this.getShareClasses.length === 0
       ? 1
       : this.getShareClasses[this.getShareClasses.length - 1].priority + 1
-    this.nextId = this.getShareClasses.length === 0
-      ? 1
-      : (this.getShareClasses.reduce((prev, current) => (prev.id > current.id) ? prev : current)).id + 1
+    this.shareId = uuidv4() // assign a new (random) ID
```

Comment:
Why not move this to EditShareStructure.vue and do away with passing the prop down?

Also on line 580 below.